executor/analyze.go

```diff
@@ -88,79 +88,31 @@ const (
 // It will collect all the sample task and run them concurrently.
 func (e *AnalyzeExec) Next(ctx context.Context, _ *chunk.Chunk) error {
 	statsHandle := domain.GetDomain(e.Ctx()).StatsHandle()
-	var tasks []*analyzeTask
-	tids := make(map[int64]struct{}) // use a set for tid
-	skippedTables := make([]string, 0)
-	is := e.Ctx().GetInfoSchema().(infoschema.InfoSchema)
-	for _, task := range e.tasks {
-		var tableID statistics.AnalyzeTableID
-		switch task.taskType {
-		case colTask:
-			tableID = task.colExec.tableID
-		case idxTask:
-			tableID = task.idxExec.tableID
-		case fastTask:
-			tableID = task.fastExec.tableID
-		case pkIncrementalTask:
-			tableID = task.colIncrementalExec.tableID
-		case idxIncrementalTask:
-			tableID = task.idxIncrementalExec.tableID
-		}
-		// skip locked tables
-		if !statsHandle.IsTableLocked(tableID.TableID) {
-			tasks = append(tasks, task)
-		}
-		// generate warning message
-		if _, ok := tids[tableID.TableID]; !ok {
-			if statsHandle.IsTableLocked(tableID.TableID) {
-				tbl, ok := is.TableByID(tableID.TableID)
-				if !ok {
-					return nil
-				}
-				skippedTables = append(skippedTables, tbl.Meta().Name.L)
-			}
-			tids[tableID.TableID] = struct{}{}
-		}
-	}
-
-	if len(skippedTables) > 0 {
-		tables := skippedTables[0]
-		for i, table := range skippedTables {
-			if i == 0 {
-				continue
-			}
-			tables += ", " + table
-		}
-		var msg string
-		if len(tids) > 1 {
-			if len(tids) > len(skippedTables) {
-				msg = "skip analyze locked tables: " + tables + ", other tables will be analyzed"
-			} else {
-				msg = "skip analyze locked tables: " + tables
-			}
-		} else {
-			msg = "skip analyze locked table: " + tables
-		}
+	infoSchema := e.Ctx().GetInfoSchema().(infoschema.InfoSchema)
```

Comment:
use ```sessiontxn.GetTxnManager(ctx).GetTxnInfoSchema``` to replace it. because this method is deprecated.

Reply:
Addressed. Thanks!